### PR TITLE
Make CorfuDB JDK 11 compatible

### PR DESCRIPTION
**Pull request by chetangudisagar**
## Overview

Description: Bring changes to make CorfuDB work with JDK 11.
cc: @xnull @PavelZaytsev 

After this PR is merged, CorfuDB will be migrated to JDK 11 and it will no longer work with JDK 8.

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: .ci/Makefile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (xnull) [inline .ci/Makefile:1]: please remove it, we don't need it

**Review by xnull — APPROVED** [review]
We had offline talk, we are good to go

File: .ci/Makefile (hunk 1 above)
Comment (chetangudisagar) [reply .ci/Makefile:1]: Done, Thanks!

**Review by PavelZaytsev — APPROVED** [review]

**Review by chetangudisagar — COMMENTED** [review]
LGTM for merging into master.

**Review by Lujie1996 — APPROVED** [review]
LGTM

**Review by hisundar — APPROVED** [review]
LGTM

**Comment by hisundar** [comment]
Please fix up the README of corfudb main page if possible to reflect that we are now on JDK 11. (The page still reads JDK 8)

**Review by xnull — APPROVED** [review]

**Review by zfrenette — APPROVED** [review]

**Comment by xnull** [comment]
Overall we age good to go, even if, unfortunately our check stuck on the 'shutdown' step. 
Universe framemork and codacy checks finished successfully, but the docker action stuck on [saving caches](https://github.com/CorfuDB/CorfuDB/actions/runs/8176072489/job/22357226357?pr=3916):
```
1h 56m 51s
Post job cleanup.
/usr/bin/sh -c docker save '4409dd4d7690' '4409dd4d7690' '448d7d602cef' '448d7d602cef' '45d9ffae5d24' '45d9ffae5d24' 'ff62239[1](https://github.com/CorfuDB/CorfuDB/actions/runs/8176072497/job/22354653437?pr=3916#step:13:1)eb9d' 'ff6[2](https://github.com/CorfuDB/CorfuDB/actions/runs/8176072497/job/22354653437?pr=3916#step:13:2)2[3](https://github.com/CorfuDB/CorfuDB/actions/runs/8176072497/job/22354653437?pr=3916#step:13:3)91eb9d' 'ce41cce20401' 'ce41cce20401' 'dee8cc22ca0c' 'dee8cc22ca0c' 'fbc7463e1d3d' 'fbc7463e1d3d' '5d64c838351a' '5d64c838351a' 'f2154f41f10d' 'f2154f41f10d' '7ac6d0d1d253' '7ac6d0d1d253' '020b706eb847' '020b706eb847' 'corfudb-universe/corfu-server:0.4.0-SNAPSHOT' '4409dd4d7690' 'corfudb/corfu-server:0.4.0-SNAPSHOT' '4409dd4d7690' 'corfudb/corfu-server:cloud' '4409dd4d7690' | tar xf - -C .
tar: d10a8d7aec24d42b35190cfadf07da8bf030b0265cbb00e925220f6926046c42/layer.tar: Wrote only 7168 of 10240 bytes
```

The satackey plugin is very old ([4 years old](https://github.com/satackey/action-docker-layer-caching/releases/tag/v0.0.11)).  I am going to replace it with the gh action provided by docker, In the next PR. The the issue will disappear.

**Pull request merged**